src/main/java/org/mamute/dao/QuestionDAO.java

```diff
@@ -278,5 +277,126 @@ public Question questionWith(Attachment attachment) {
 				.setParameter("a", attachment)
 				.uniqueResult();
 	}
-}
 
+	public void delete(Question question) {
+		session.createSQLQuery("SET foreign_key_checks = 0").executeUpdate();
+		QuestionQuery query = new QuestionQuery(question);
+
+		query.execute("delete f,qf from Question q" +
+				" join Question_Flags qf on qf.Question_id=q.id" +
+				" join Flag f on f.id=qf.flags_id" +
+				" where q.id=:id");
+		query.execute("delete w,qw from Question q" +
+				" join Question_Watchers qw on qw.Question_id=q.id" +
+				" join Watcher w on w.id=qw.watchers_id" +
+				" where q.id=:id");
+		query.execute("delete from ReputationEvent where context_id=:id and context_type='QUESTION'");
+
+		deleteAttachments(question);
+		deleteRelationshipTables(question);
+		session.createSQLQuery("delete from QuestionInformation_Tag where QuestionInformation_id=:id")
+				.setParameter("id", question.getInformation().getId())
+				.executeUpdate();
+		query.execute("delete from QuestionInformation where question_id=:id");
+		query.execute("delete from Question where id=:id");
+		session.createSQLQuery("SET foreign_key_checks = 1").executeUpdate();
+	}
+
+	private void deleteRelationshipTables(Question question) {
+		List<String> relationshipTables = Arrays.asList("Question_Interactions");
```

Comment:
I was deleting data from other join tables in this method before. Now it doesn't make sense anymore, I'll refactor.